lib/pbench/server/api/resources/upload_api.py

```diff
@@ -423,8 +429,41 @@ def finalize_dataset(
             except Exception as e:
                 logger.error("Unable to get integer retention days: {}", str(e))
                 raise
+
+            # Now that we have the tarball, extract the dataset timestamp from
+            # the metadata.log file.
+            #
+            # If this fails, the metadata.log is missing or corrupt, so fail
+            # the upload.
+            #
+            # TODO: If we did this before the `advance` we could avoid a second
+            # `update` operation, though that means relying on the implicit
+            # update in `advance`, which is perhaps a bit weird.
```

Comment:
I can definitely see an argument for pulling a file out of the tarball as a separate method, probably in `Tarball`; because our dynamic unpacking cache will need that anyway. I don't think the `Dataset` mutation (much less the commit) should be pulled with it, though; that's uniquely "PUT".